### PR TITLE
CTCP-4308  add Departure transport means to CYA

### PR DESCRIPTION
**Pull request by kenneth-rayner**

### CODE REVIEW

File: app/controllers/transport/departureTransportMeans/TransportMeansIdentificationController.scala [hunk 1]
```diff
@@ -78,18 +86,17 @@ class TransportMeansIdentificationController @Inject() (
...
   private def redirect(
-    mode: Mode,
     value: TransportMeansIdentification,
     departureId: String
   )(implicit request: MandatoryDataRequest[_]): Future[Result] =
     for {
       updatedAnswers <- Future.fromTry(request.userAnswers.set(TransportMeansIdentificationPage, value))
       _              <- sessionRepository.set(updatedAnswers)
-    } yield Redirect(navigator.nextPage(TransportMeansIdentificationPage, updatedAnswers, departureId, mode))
+    } yield Redirect(controllers.routes.CheckYourAnswersController.onPageLoad(departureId))
```
Comment (jackheslop96) [inline app/controllers/transport/departureTransportMeans/TransportMeansIdentificationController.scala:101]: Would prefer to keep navigation logic within the navigators if possible

File: app/models/messages/DepartureTransportMeans.scala [hunk 2]
```diff
@@ -19,6 +19,7 @@ package models.messages
 import play.api.libs.json.{Json, OFormat}
 
 case class DepartureTransportMeans(
+  sequenceNumber: String,
```
Comment (jackheslop96) [inline app/models/messages/DepartureTransportMeans.scala:22]: Is this being used? I appreciate it is part of the model in the DDNTA but if we don't use it then the reads won't be affected by omitting it from the case class

File: app/models/messages/Consignment.scala [hunk 3]
```diff
@@ -25,13 +25,14 @@ case class Consignment(
   inlandModeOfTransport: Option[String],
   TransportEquipment: Option[List[TransportEquipment]],
   LocationOfGoods: Option[LocationOfGoods],
-  DepartureTransportMeans: Option[DepartureTransportMeans],
+  DepartureTransportMeans: Option[List[DepartureTransportMeans]],
```
Comment (jackheslop96) [inline app/models/messages/Consignment.scala:28]: There looks to be a lot of changes related to this. Could it be worth exploring changing this back to `Option[DepartureTransportMeans]` and defining the reads to read it in as a list of DepartureTransportMeans and take the head option?

File: app/models/messages/DepartureTransportMeans.scala (hunk 2 above)
Comment (kenneth-rayner) [reply app/models/messages/DepartureTransportMeans.scala:22]: Done

File: app/controllers/transport/departureTransportMeans/TransportMeansIdentificationController.scala (hunk 1 above)
Comment (kenneth-rayner) [reply app/controllers/transport/departureTransportMeans/TransportMeansIdentificationController.scala]: I had considered that , but my thinking was that the 3 pages on this ticket only get landed on from CYA change links and always go back to CYA, so was there a necessity to create a new Navigator just for 3 pages to go back to CYA.
However, I have now added  this for consistency

**Review by jackheslop96 — APPROVED** [review]

**Pull request merged**